libraries/net/node.cpp

```diff
@@ -335,6 +321,44 @@ namespace graphene { namespace net { namespace detail {
       std::vector<graphene::net::address_info> advertise_list;
    };
 
+   /****
+    * Advertise all nodes except a predefined list
+    */
+   class exclude_address_builder : public node_impl::address_builder
+   {
+      public:
+      exclude_address_builder(fc::optional<std::vector<std::string>> address_list)
+      {
+         if (address_list.valid())
+         {
+            exclude_list = *address_list;
+         }
+      }
+      void build(node_impl* impl, address_message& reply)
+      {
+         reply.addresses.reserve(impl->_active_connections.size());
+         for(const peer_connection_ptr& active_peer : impl->_active_connections)
+         {
+            if (!exists_in_list(active_peer, exclude_list))
+               reply.addresses.emplace_back(update_address_record(impl, active_peer));
+         }
+         reply.addresses.shrink_to_fit();
+      }
+      private:
+      std::vector<std::string> exclude_list;
```

Comment:
Better use a `flat_set` here, then you can do lookups using `find` below, which is more efficient.

Reply:
Care to wager? :-D I have not tested it, but am betting the linear search of vector outperforms flat_set here, unless we are talking 1000's of entries. I will check.

Reply:
Agree that it won't make much of a difference performance-wise, but it does simplify the code.

Reply:
Done